xml2rfc_compat/tests/test_adapters.py

```diff
@@ -139,3 +142,17 @@ def test_nist_not_found(self):
         adapter = NistAdapter(self.dirname, "bibxml-nist", self.nist_ref+"A")
         with self.assertRaises(RefNotFoundError):
             adapter.resolve()
+
+    def test_doi(self):
+        adapter = DoiAdapter(self.dirname, "bibxml7", self.doi_ref)
+        bibitem = adapter.resolve()
+        self._assert_is_instance_of_bibliographicitem(bibitem)
+        self._assert_refs_equal(bibitem, self.doi_ref)
+
+    def test_doi_should_replace_target_URL(self):
+        adapter = DoiAdapter(self.dirname, "bibxml7", self.doi_ref)
+        bibitem = adapter.resolve()
+        for link in as_list(bibitem.link or []):
+            if content := link.__getattribute__("content"):
+                url_parse = urlparse(content)
+                self.assertNotEqual("http://dx.doi.org", f"{url_parse.scheme}://{url_parse.netloc}")
```

Comment:
Isn't it better to test if `f"{url_parse.scheme}://{url_parse.netloc}"` equals to `https://doi.org`?

Reply:
There might be more than one Link (BibliographicItem.link it can be a list, see [here](https://github.com/relaton/relaton-py/blob/c2ab40f925ec1ca394851ff5746fdcb3b8258499/relaton/models/bibdata.py#L182)).

As far as I am aware, only one of the Link instance has content http://dx.doi.org (then converted to https://doi.org). The others can have any value. @kesara

Reply:
Thus the assertion for not equality, rather than for equality

Reply:
Perhaps then check if at least one URL starts with `http://dx.doi.org`? Because checking for not equal doesn't very that the correct doi URL is present.